spec/appendices.md

```diff
@@ -0,0 +1,33 @@
+# DRAFT Appendices
+
+## Security Considerations
+
+MessageFormat 2.0 _patterns_ are meant to allow a _message_ to include any string value
+which users might normally wish to use in their environment.
+Programming languages and other environments vary in what characters are permitted
+to appear in a valid string.
+In many cases, certain types of characters, such as invisible control characters,
+require escaping by these host formats.
+In other cases, strings are not permitted to contain certain characters at all.
+Since _messages_ are subject to the restrictions and limitations of their 
+host environments, their serializations and resource formats,
+that might be sufficient to prevent most problems.
+However, MessageFormat itself does not supply such a restriction.
+
+MessageFormat _messages_ permit nearly all Unicode code points,
+with the exception of surrogates, to appear in a _pattern_.
+This means that it can be possible for a _message_ to contain invisible unquoted
+control characters that abnormally affect the display of the _message_
+but do not generate errors from MessageFormat parsers or processing APIs.
+
+Bidirectional text containing right-to-left characters (such as used for Arabic or Hebrew) 
+also poses a potential source of confusion for users. 
+Since MessageFormat 2.0's syntax makes use of 
```

Comment:
I think we need to separate 2 cases: what the MF message appears like to the programmer, vs what a resulting message appears like to the user. 

1. For the former, UTS55 is relevant.
2. For the latter, we can point out that MF conformance requires that placeholder replacements be generated in a way that avoids bidi reordering issues between the literal message text and the placeholder contents.

Reply:
This text refers specifically to viewing the _message_ in a source code context (and thus UTS55 is relevant). 

That reminds me that there is probably the need to call out another security consideration, which is that the output of formatting needs to be examined for e.g. injection or other abuse.